autogen.sh

```diff
@@ -1,2 +1,6 @@
-#!/bin/sh
-autoreconf -i
+#!/bin/bash -e
```

Comment:
I'd rather keep it using shell, not all environments have bash and the script does not seem to use anything bash-specific.

You can add flags using `set -e` in a new line, which seems to be more common @Ced

Reply:
When using BASH it is common to use both `#!/bin/bash -e` or `#!/bin/bash set -e`. As far as I know, there is no standard/recommendation about them. I preferred the first one because it is more compact.

As @ftynse stated, there is no need to use BASH instead of SHELL since there is nothing bash-specific. My bad, I just didn't think about other environments not supporting BASH. 
However, I consider important to make the script fail whenever an error occurs on executing the internal commands because otherwise it gets complicated to handle errors when installing it in a larger pipeline.

Reply:
Reverted to `#!/bin/sh` and `set -e`.